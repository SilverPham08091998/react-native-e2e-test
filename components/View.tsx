import { View as DefaultView } from 'react-native';
import { useThemeColor, ThemeProps } from './Themed';

export type ViewProps = ThemeProps & DefaultView['props'];

export function View(props: ViewProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'background');

  return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
}
