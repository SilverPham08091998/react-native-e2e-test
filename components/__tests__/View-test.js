import * as React from 'react';
import renderer from 'react-test-renderer';

import { View } from '../View';

it(`renders correctly`, () => {
  const onPress = jest.fn()
  const tree = renderer.create(<View style={{ width: 100, height: 100 }} lightColor='red' darkColor='green' />).toJSON();

  expect(tree).toMatchSnapshot();
});
