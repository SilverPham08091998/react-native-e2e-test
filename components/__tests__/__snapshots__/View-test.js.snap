// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "red",
      },
      {
        "height": 100,
        "width": 100,
      },
    ]
  }
/>
`;
